packages/cardano-graphql/src/DataProjection/types.ts

```diff
@@ -32,16 +33,68 @@ export interface Upsert {
   };
 }
 
-// export interface CombinedQueryResult {}
+export interface CombinedQueryResult {
+  data: {
+    assets: Asset[];
+  };
+}
+
+export interface ProcessingResult {
+  assets: Asset[];
+}
+
+export interface CombinedProcessingResult {
+  func: void | Promise<Partial<ProcessingResult>>;
+  id: string;
+}
 
-// export interface CombinedProcessingResult {}
+export interface ProcessParameters {
+  ctx?: RollForwardContext;
+  queryResult?: CombinedQueryResult;
+}
 
 export interface BlockHandler {
   id: string;
   query?: (ctx: RollForwardContext) => Promise<QueryResult<QueryVariables>>;
-  // process?: (ctx: RollForwardContext, queryResult?: CombinedQueryResult) => Promise<Partial<CombinedProcessingResult>>;
-  process?: (ctx: RollForwardContext, queryResult?: any) => Promise<Partial<any>>;
+  process?: (parameters: ProcessParameters) => Promise<Partial<ProcessingResult>>;
   rollBackward: (ctx: RollBackwardContext) => Promise<Upsert>;
-  // rollForward: (ctx: RollForwardContext, processingResult: CombinedProcessingResult) => Promise<Upsert>;
-  rollForward: (ctx: RollForwardContext, processingResult?: any) => Promise<Upsert>;
+  rollForward: (ctx: RollForwardContext, processingResult: CombinedProcessingResult[]) => Promise<Upsert>;
+}
+
+export interface Signature {
+  signature: string;
+  publicKey: string;
+}
+
+export interface AssetMetadata {
```

Comment:
Yes, `MetadataClient` isn't exclusively for data projection, so when this code is moved to a new `cardano-graphql-services` package, it can live in the source root.